policy/release/olm.rego

```diff
@@ -215,6 +215,49 @@ deny contains result if {
 	result := lib.result_helper_with_term(rego.metadata.chain(), [image.str(unmatched_image.ref)], image.str(unmatched_image.ref))
 }
 
+# METADATA
+# title: Images referenced by OLM bundle are from allowed registries
+# description: >-
+#   Images referenced by the OLM bundle are from allowed registries
+# custom:
+#   short_name: allowed_registries
+#   failure_msg: The %q CSV image reference is not from an allowed registry.
+#   solution: >-
+#     The image must be hosted on an allowed registry.
+#   collections:
+#   - redhat
+#   effective_on: 2024-08-15T00:00:00Z
+#
+deny contains result if {
+	# Restrict to release pipeline for now
+	_release_restrictions_apply
+
+	# The presence of expected rule_data verified in _rule_data_errors
+	allowed_registry_prefixes := lib.rule_data("allowed_registry_prefixes")
+
+	# Parse manifests from snapshot
+	some csv_manifest in _csv_manifests
+
+	# Parse image references from each manifest
+	all_csv_images := all_image_ref(csv_manifest)
+
+	# Filter out paths for set of unique image references
+	unique_image_refs := {img_ref |
+		some img in all_csv_images
+		img_ref = img.ref
+	}
+
+	disallowed_registry_image_refs := [img_ref |
+		some img_ref in unique_image_refs
+		not _image_registry_allowed(img_ref.repo, allowed_registry_prefixes)
+	]
+
+	some disallowed_reg_image in disallowed_registry_image_refs
+
+	# regal ignore:line-length
+	result := lib.result_helper_with_term(rego.metadata.chain(), [image.str(disallowed_reg_image)], image.str(disallowed_reg_image))
```

Comment:
I think for the term, the last parameter, we probably want to omit any tags and digests. Only include the repo, e.g. quay.io/foo/bar. This could then be used to create a certain policy exclusion.

wdyt?

Reply:
Good idea... it would make copy/paste easier